mythril/analysis/modules/delegatecall.py

```diff
@@ -12,57 +19,77 @@ def __init__(self):
+            entrypoint="callback",
         )
+        self._issues = []
 
-    def execute(self, statespace):
-        """
-        Executes analysis module for delegate call analysis module
-        :param statespace: Statespace to analyse
-        :return: Found issues
-        """
-        issues = []
+    @property
+    def issues(self) -> list:
+        return self._issues
 
-        for call in statespace.calls:
+    def execute(self, state: GlobalState) -> list:
+        log.debug("Executing module: DELEGATE_CALL")
+        self._issues.extend(_analyze_states(state))
+        return self.issues
 
-            if call.type is not "DELEGATECALL":
-                continue
-            if call.node.function_name is not "fallback":
-                continue
 
-            state = call.state
-            address = state.get_current_instruction()["address"]
-            meminstart = get_variable(state.mstate.stack[-3])
+def _analyze_states(state: GlobalState) -> list:
+    """
+    :param state: the current state
+    :return: returns the issues for that corresponding state
+    """
+    call = get_call_from_state(state)
+    issues = []
 
-            if meminstart.type == VarType.CONCRETE:
-                issues += self._concrete_call(call, state, address, meminstart)
+    if call.type is not "DELEGATECALL":
+        return []
+    if call.node.function_name is not "fallback":
+        return []
 
-        return issues
+    state = call.state
+    address = state.get_current_instruction()["address"]
+    meminstart = get_variable(state.mstate.stack[-3])
 
-    def _concrete_call(self, call, state, address, meminstart):
-        if not re.search(r"calldata.*_0", str(state.mstate.memory[meminstart.val])):
-            return []
+    if meminstart.type == VarType.CONCRETE:
+        issues += _concrete_call(call, state, address, meminstart)
 
-        issue = Issue(
-            contract=call.node.contract_name,
-            function_name=call.node.function_name,
-            address=address,
-            swc_id=DELEGATECALL_TO_UNTRUSTED_CONTRACT,
-            bytecode=state.environment.code.bytecode,
-            title="Call data forwarded with delegatecall()",
-            _type="Informational",
-            gas_used=(state.mstate.min_gas_used, state.mstate.max_gas_used),
-        )
+    return issues
 
-        issue.description = (
-            "This contract forwards its call data via DELEGATECALL in its fallback function. "
-            "This means that any function in the called contract can be executed. Note that the callee contract will have "
-            "access to the storage of the calling contract.\n"
-        )
 
-        target = hex(call.to.val) if call.to.type == VarType.CONCRETE else str(call.to)
-        issue.description += "DELEGATECALL target: {}".format(target)
+def _concrete_call(
+    call: Call, state: GlobalState, address: int, meminstart: Variable
+) -> list:
```

Comment:
```suggestion
) -> List[Issue]:
```
and on top we'll have to `from typing import List` :slightly_smiling_face: